sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/AbstractSparkSQLParser.scala

```diff
@@ -129,7 +129,7 @@ class SqlLexical extends StdLexical {
   override def identChar: Parser[Elem] = letter | elem('_')
 
   private lazy val scientificNotation: Parser[String] =
-    (elem('e') | elem('E')) ~> (elem('+') | elem('-')).? ~ rep1(digit) ^^ {
+    (elem('e') | elem('E')) ~> (elem('+') | elem('-')).? ~ rep1(digit) <~ not(rep1(identChar)) ^^ {
```

Comment:
The `SqlLexical` is still used in the `DataTypeParser`. About 50% of this class is redundant.